### ISSUE TITLE
Transiently Failing Selenium Tests

### ISSUE THREAD

**Issue by jmchilton** [galaxyproject/galaxy#3838]
We have an increasing number of Selenium tests and most pass most of the time, but about half the time one test in the suite will fail.
 
![image](https://cloud.githubusercontent.com/assets/216771/24453034/310a47e0-1454-11e7-9653-b4f195aff07f.png)

- test_history_sharing.py:test_sharing_valid - This test features two different users logging in and it seems like there is some finicky-ness related to this. I may be worth just re-running this whole test on failure.
  - https://jenkins.galaxyproject.org/view/All/job/selenium/137/testReport/junit/selenium_tests.test_history_sharing/HistorySharingTestCase/test_sharing_valid/
  - https://jenkins.galaxyproject.org/view/All/job/selenium/117/testReport/selenium_tests.test_history_sharing/HistorySharingTestCase/test_sharing_valid/
  - https://jenkins.galaxyproject.org/view/All/job/selenium/115/testReport/junit/selenium_tests.test_history_sharing/HistorySharingTestCase/test_sharing_valid/

- test_workflow_run.py:test_simple_execution - I've seen this one fail multiple times - I guess the problem is waiting on datasets in the side bar to appear. I think given that the GUI seems to just not update for long periods of time it would be worth manually refreshing while waiting for datasets every say 15 seconds.
  - https://jenkins.galaxyproject.org/view/All/job/selenium/131/testReport/junit/selenium_tests.test_workflow_run/WorkflowRunTestCase/test_simple_execution/
  - https://jenkins.galaxyproject.org/view/All/job/selenium/126/testReport/junit/selenium_tests.test_workflow_run/WorkflowRunTestCase/test_simple_execution/
  - https://jenkins.galaxyproject.org/view/All/job/selenium/112/testReport/junit/selenium_tests.test_workflow_run/WorkflowRunTestCase/test_simple_execution/

Things I haven't seen more than once:
 - https://jenkins.galaxyproject.org/view/All/job/selenium/128/testReport/junit/selenium_tests.test_tool_form/ToolFormTestCase/test_rerun/ (rerun failing to render the form properly?)
 -  https://jenkins.galaxyproject.org/view/All/job/selenium/127/testReport/junit/selenium_tests.test_workflow_editor/WorkflowEditorTestCase/test_editor_invalid_tool_state/ (workflow editor error pop-up not rendering the correct message?)

xref #3361 
xref #1617

**Comment by jmchilton** [comment]
This is pretty dated at this point - I'm going to start a new issue with fresh failures.

**Comment by martenson** [comment]
great job with the test!

![screenshot 2017-09-22 10 35 51](https://user-images.githubusercontent.com/1814954/30749589-dbc73c16-9f81-11e7-8652-fa5150bef6d6.png)
